src/store/store.ts

```diff
@@ -0,0 +1,11 @@
+import { configureStore } from "@reduxjs/toolkit"
+import counterSlice from "./counterSlice"
+
+export const store = configureStore({
+  reducer: {
+    counter: counterSlice
+  },
+})
+
+export type RootState = ReturnType<typeof store.getState>
```

Comment:
Can you give me a few more details here - why we do it?

Reply:
the same as above, separate types for handy using
https://redux-toolkit.js.org/tutorials/typescript#define-root-state-and-dispatch-types